test-js/test-utils.js

```diff
@@ -0,0 +1,211 @@
+      [1, 2],
+      [0, 1],
+    ]);
+  });
+});
+
+describe("range", () => {
+  it("should generate a range of numbers", () => {
+    const numbers = [...range(3)];
+    expect(numbers).deep.equals([0, 1, 2]);
+  });
+  it("should generate a range of numbers, with start and stop values", () => {
+    const numbers = [...range(10, 13)];
+    expect(numbers, [10, 11, 12]);
+  });
+  it("should generate a range of numbers, with start, stop and step values", () => {
+    const numbers = [...range(10, 15, 2)];
+    expect(numbers, [10, 12, 14]);
+  });
+});
+
+describe("chain", () => {
+  it("chain iterators", () => {
+    const chained = [...chain(range(2), range(2))];
+    expect(chained, [0, 1, 0, 1]);
+  });
+});
+
+describe("makeUPlusStringFromCodePoint", () => {
+  it("throws an exception when an invalid parameter is given", () => {
+    expect(() => makeUPlusStringFromCodePoint("not-a-number")).to.throw();
+  });
+  it("should not throw an exception for a falsy value", () => {
+    expect(() => makeUPlusStringFromCodePoint("")).to.not.throw();
+  });
+  it("make a number unicode hex", () => {
+    expect(makeUPlusStringFromCodePoint(97)).equals("U+0061"); // a
+    expect(makeUPlusStringFromCodePoint(65)).equals("U+0041"); // A
+  });
+});
+
+describe("getCharFromUnicode", () => {
+  it("should return an empty string if an argument is not passed", () => {
+    expect(getCharFromUnicode()).equals("");
+  });
+  it("should convert a unicode symbol number to a character", () => {
+    expect(getCharFromUnicode(97)).equals("a");
+  });
+});
+
+describe("guessCharFromGlyphName", () => {
+  it("should guess a character from code points in free text", () => {
+    expect(guessCharFromGlyphName("text 0061 text")).equals("a");
+    expect(guessCharFromGlyphName("text ff0061 text")).equals("a");
+    expect(guessCharFromGlyphName("text ff0041 text")).equals("A");
+    expect(guessCharFromGlyphName("text 110000 text")).equals("");
+    expect(guessCharFromGlyphName("text 10FFFF text")).equals("");
+    expect(guessCharFromGlyphName("text 100000 text")).equals("");
+    expect(guessCharFromGlyphName("text 1F440 text")).equals("👀");
+  });
+});
+
+describe("fileNameExtension", () => {
+  it("should return the file extension of a file name", () => {
+    expect(fileNameExtension("test-utils.js")).equals("js");
+  });
+  it("should work well when there is a dot in the file name", () => {
+    expect(fileNameExtension("test.utils.js")).equals("js");
+  });
+  it("should return the given file name when there is no extension", () => {
+    expect(fileNameExtension("utils")).equals("utils");
+  });
+});
+
+describe("arrayExtend", () => {
+  it("should extend arrays with another one", () => {
+    const array = [1, 2, 3];
+    arrayExtend(array, [4, 5]);
+    expect(array).deep.equals([1, 2, 3, 4, 5]);
+  });
+  it("test chunk-by-chunk addition by 1024", () => {
+    const destinationArray = [1, 2, 3];
+    arrayExtend(destinationArray, [...range(1024)]);
```

Comment:
We need to test _more than the chunk size_ to trigger the multiple chunk code path. That's why I suggested to do 3000. 1025 would be enough, actually.

Reply:
Ups, missed that, sent one more commit.

Reply:
Btw tests passes when I replace the arrayExtend content with `thisArray.push(...itemsArray);`. I'm curious, why it's needed to add them separately on each 1024?